libs/libmysqlxx/src/Connection.cpp

```diff
@@ -88,7 +90,7 @@ void Connection::connect(const char* db,
     if (mysql_options(driver.get(), MYSQL_OPT_LOCAL_INFILE, nullptr))
         throw ConnectionFailed(errorMessage(driver.get()), mysql_errno(driver.get()));
 
-    if (!mysql_real_connect(driver.get(), server, user, password, db, port, nullptr, driver->client_flag))
+    if (!mysql_real_connect(driver.get(), server, user, password, db, port, *socket ? socket : nullptr, driver->client_flag))
```

Comment:
If server is `localhost`, then port is ignored and socket used. Otherwise, socket is ignored.
Similar behavior has linux mysql client.